packages/cli/lib/artifact-generator.js

```diff
@@ -19,23 +19,58 @@ module.exports = class ArtifactGenerator extends BaseGenerator {
 
   _setupGenerator() {
     debug('Setting up generator');
+    super._setupGenerator();
     this.argument('name', {
       type: String,
       required: false,
       description: 'Name for the ' + this.artifactInfo.type,
     });
+  }
+
+  setOptions() {
     // argument validation
-    if (this.args.length) {
-      const validationMsg = utils.validateClassName(this.args[0]);
+    const name = this.options.name;
+    if (name) {
+      const validationMsg = utils.validateClassName(name);
       if (typeof validationMsg === 'string') throw new Error(validationMsg);
     }
-    this.artifactInfo.name = this.args[0];
-    this.artifactInfo.defaultName = 'new';
+    this.artifactInfo.name = name;
     this.artifactInfo.relPath = path.relative(
       this.destinationPath(),
       this.artifactInfo.outDir,
     );
-    super._setupGenerator();
+    return super.setOptions();
+  }
+
+  /**
+   * Checks if current directory is a LoopBack project by checking for
+   * keyword 'loopback' under 'keywords' attribute in package.json.
+   * 'keywords' is an array
+   */
+  checkLoopBackProject() {
```

Comment:
Don't we already have another implementation of `checkLoopBackProject`? I think @virkt25 was using one in his datasource work. Let's create a single implementation of this check and move it to a place where all generators can import it from.

Reply:
@raymondfeng PTAL ☝️